libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -273,6 +272,25 @@ void AP_AIS::send(mavlink_channel_t chan)
     }
 }
 
+void AP_AIS::handle_message(const mavlink_ais_vessel_t &packet)
+{
+    if (!enabled()) {
+        return;
+    }
+
+    uint16_t index;
+    if (!get_vessel_index(packet.MMSI, index)) {
+        // not found and not enough memory to add it
+        return;
+    }
+    memcpy(&_list[index].info, &packet, sizeof(_list[index].info));
+    _list[index].last_update_ms = AP_HAL::millis() - packet.tslc * 1000;
```

Comment:
Shouldn't we check that the "new" data is actually newer than what we have? If we do that and round tslc up instead of down when sending AIS_vessel frames we should prevent falsely increasing last_update_ms and having infinite ping-pong.

```suggestion
    if(_list[index].last_update_ms < AP_HAL::millis() - packet.tslc * 1000) {
        memcpy(&_list[index].info, &packet, sizeof(_list[index].info));
        _list[index].last_update_ms = AP_HAL::millis() - packet.tslc * 1000;
    }
```    
In send function:
```
_list[_send_index].info.tslc = (now - _list[_send_index].last_update_ms+999) * 0.001;
```
This way we will "decay" messages on rebroadcasting.

Reply:
@LupusTheCanine I know I know.. but what's more important right now is it doesn't mess up any comms and this is implemented exactly how ADSB is which is a known-good configuration. I would prefer this to go in as-is now and we can fix both this and ADSB later in a future PR